impeller/entity/save_layer_utils.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/entity/save_layer_utils.h"
+
+namespace impeller {
+
+std::optional<Rect> ComputeSaveLayerCoverage(
+    const Rect& content_coverage,
+    const Matrix& effect_transform,
+    const Rect& coverage_limit,
+    const std::shared_ptr<FilterContents>& image_filter,
+    bool flood_output_coverage,
+    bool flood_input_coverage) {
+  Rect coverage = content_coverage;
+  // There are two conditions that currently trigger flood_input_coverage, the
+  // first is the presence of a backdrop filter and the second is the presence
+  // of a color filter that effects transparent black.
```

Comment:
I think you are talking about a CF or BDF on this SL itself?

Note that also a CF or a BDF on a child SL within the content of this SL can also flood. The options have a "content_is_unbounded" flag for that case.

Reply:
Added some clarifications, and the third case which has already been handled where the SL contains unbounded content.